lib/private/legacy/OC_Util.php

```diff
@@ -1100,7 +1100,7 @@ public static function getDefaultPageUrl() {
 			} else {
 				$appId = 'files';
 				$config = \OC::$server->getConfig();
-				$defaultApps = explode(',', $config->getSystemValue('defaultapp', 'files'));
+				$defaultApps = explode(',', $config->getSystemValue('defaultapp', 'dashboard'));
```

Comment:
```suggestion
				$defaultApps = explode(',', $config->getSystemValue('defaultapp', 'dashboard,files'));
```

Then it would use the first one that is enabled of those two and allows the dashboard app being disabled and still having a default app.